### PR TITLE
Resolves #214: Restrict some characters from node names 

### LINKED ISSUE #214: Restrict some characters from node names
Do not allow files and folders names to contain characters such as '/' and '\\'.

### PR DESCRIPTION
**Pull request by robertmercea**
Solves issue #214 
Added name-validating functions in `/src/mount/linux.rs` and `/src/encryptedfs.rs`.
Use said functions in multiple methods to check for `'/'` and `'\'` characters.

### CODE REVIEW

File: src/encryptedfs.rs [hunk 1]
```diff
@@ -657,6 +657,13 @@ impl EncryptedFs {
         self.read_only
     }
 
+    fn validate_filename(&self, filename: &str) -> FsResult<bool> {
+        if filename.contains('/') || filename.contains('\\') {
+            return Err(FsError::InvalidInput("invalid file name"));
```
Comment (radumarias) [inline src/encryptedfs.rs]: Have two conditions and explicitly say in the error message what char is invalid, like
```
if filename.contains('/') {
    Err(FsError::InvalidInput("'/' not allowed in the filename"))
} else if filename.contains('/') {
    Err(FsError::InvalidInput("'\\' not allowed in the filename"))
} else {
    Ok(())
}
```

You don't need ret `bool` as you ret Err, just return nothing `FsResult<()>`.

**Review by radumarias — CHANGES_REQUESTED** [review]

File: src/encryptedfs.rs [hunk 2]
```diff
@@ -1998,6 +2015,7 @@ impl EncryptedFs {
         if !self.exists_by_name(parent, name)? {
             return Err(FsError::NotFound("name not found"));
         }
+        self.validate_filename(&new_name.expose_secret().to_string())?;
```
Comment (radumarias) [inline src/encryptedfs.rs:2018]: if it's called in all places from SecretVec better have param as SecretVec

File: src/mount/linux.rs [hunk 3]
```diff
@@ -153,6 +153,14 @@ impl EncryptedFsFuse3 {
         (mode & !(libc::S_ISUID | libc::S_ISGID)) as u16
     }
 
+    fn validate_filename(&self, filename: &OsStr) -> Result<bool> {
```
Comment (radumarias) [inline src/mount/linux.rs:156]: we don't need here, it's enough in EncryptedFs

**Review by radumarias — CHANGES_REQUESTED** [review]

**Comment by radumarias** [comment]
@Ropler6 any updates on this?

**Comment by robertmercea** [comment]
Will start working on it tomorrow

**Pull request closed without merging**